tools/make_opcodes.mll

```diff
@@ -35,16 +36,22 @@ and opnames = parse
       [
         "-opnames", Arg.Set print_opnames, " Dump opcode names";
         "-opcodes", Arg.Set print_opcodes, " Dump opcode numbers";
+        "-opcodes-mli", Arg.Set print_opcodes_mli,
+          " Produce the .mli for opcodes.ml";
       ]
     in
     Arg.parse (Arg.align spec) ignore "Extract opcode info from instruct.h";
     let lexbuf = Lexing.from_channel stdin in
     let id, opnames = find_enum lexbuf in
-    if !print_opnames then begin
-      printf "let names_of_%s = [|\n" id;
-      List.iter (fun s -> printf "  %S;\n" s) opnames;
-      printf "|]\n"
-    end;
-    if !print_opcodes then
-      List.iteri (fun i op -> printf "let op%s = %i\n" op i) opnames
+    if !print_opcodes_mli then begin
+      List.iteri (fun i op -> printf "val op%s : int\n" op) opnames
+    end else begin
+      if !print_opnames then begin
+        printf "let names_of_%s = [|\n" id;
+        List.iter (fun s -> printf "  %S;\n" s) opnames;
+        printf "|]\n"
+      end;
+      if !print_opcodes then
+        List.iteri (fun i op -> printf "let op%s = %i\n" op i) opnames
+    end
```

Comment:
For symmetry with existing code, it would be better to write
```
if !print_opcodes_mli then
  List.iteri ...;
if !print_opnames then begin
  ...
```